nanshe_ipython.ipynb

```diff
@@ -545,6 +546,255 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "\n",
+        "\n",
+        "num_reps = 5\n",
+        "tmpl_hist_wght = 0.25\n",
+        "thld_rel_dist = 0.0\n",
+        "\n",
+        "block_frames = 1\n",
+        "norm_frames = 100\n",
+        "\n",
+        "\n",
+        "# Somehow we can't overwrite the file in the container so this is needed.\n",
+        "io_remove(data_basename + postfix_reg + zarr_ext)\n",
+        "io_remove(data_basename + postfix_reg + h5_ext)\n",
+        "\n",
+        "\n",
+        "with open_zarr(data_basename + postfix_dn + zarr_ext, \"r\") as f:\n",
+        "    with get_executor(client) as executor:\n",
+        "        # Load and prep data for computation.\n",
+        "        imgs = f[\"images\"]\n",
+        "        da_imgs = da.from_array(imgs, chunks=imgs.chunks)\n",
+        "        da_imgs = da_imgs.rechunk((block_frames,) + imgs.shape[1:])\n",
+        "\n",
+        "        da_imgs_flt = da_imgs\n",
+        "        if not (issubclass(da_imgs_flt.dtype.type, np.floating) and \n",
+        "                da_imgs_flt.dtype.itemsize >= 4):\n",
+        "            da_imgs_flt = da_imgs_flt.astype(np.float32)\n",
+        "\n",
+        "        # Create frame array\n",
+        "        frame_shape = da.from_array(\n",
+        "            np.array(da_imgs_flt.shape[1:], dtype=int),\n",
+        "            chunks=(da_imgs_flt.ndim - 1,)\n",
+        "        )\n",
+        "\n",
+        "        # Persist frame shape\n",
+        "        frame_shape = frame_shape.persist()\n",
+        "\n",
+        "        # Compute the FFT of frames\n",
+        "        da_imgs_fft = fftn(da_imgs_flt, axes=tuple(irange(1, imgs.ndim)))\n",
+        "\n",
+        "        # Persist FFT of frames\n",
+        "        da_imgs_fft = da_imgs_fft.persist()\n",
+        "\n",
+        "        # Initialize\n",
+        "        i = 0\n",
+        "        avg_rel_dist = 1.0\n",
+        "        tmpl_hist_wght = da_imgs_flt.dtype.type(tmpl_hist_wght)\n",
+        "        shifts = da.zeros(\n",
+        "            (len(da_imgs_flt), da_imgs_flt.ndim - 1),\n",
+        "            dtype=int,\n",
+        "            chunks=(1, da_imgs_flt.ndim - 1)\n",
+        "        )\n",
+        "        da_imgs_fft_tmplt = da_imgs_fft.mean(axis=0)\n",
+        "\n",
+        "        # Persist FFT template image\n",
+        "        da_imgs_fft_tmplt = da_imgs_fft_tmplt.persist()\n",
+        "\n",
+        "        while avg_rel_dist > thld_rel_dist and i < num_reps:\n",
+        "            # Compute the shifted frames\n",
+        "            shifted_frames = []\n",
+        "            for j in irange(len(da_imgs_fft)):\n",
+        "                shifted_frames.append(dask_ndfourier.fourier_shift(\n",
+        "                    da_imgs_fft[i], shifts[i]\n",
```

Comment:
These should not be indexed by `i` as that tracks how many iterations have been performed. These should be indexed by `j` as that represents, which frame we are finding the shift of.

Reply:
Fixing in PR ( https://github.com/nanshe-org/nanshe_workflow/pull/138 ).